uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/logging/appender/FileAppender.java

```diff
@@ -0,0 +1,131 @@
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.logging.appender;
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.test.core.app.ApplicationProvider;
+
+import com.microsoft.identity.client.ui.automation.logging.LogLevel;
+import com.microsoft.identity.client.ui.automation.logging.formatter.ILogFormatter;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+/**
+ * An implementation of {@link IAppender} to write logs to a file on desk. The filename must be
+ * supplied to the FileLogger and the file will be created in the files directory reserved by the OS
+ * for the calling application. This directory is what is returned by {@link Context#getFilesDir()}.
+ */
+public class FileAppender implements IAppender {
+
+    private static final String TAG = FileAppender.class.getSimpleName();
+
+    private final String mFileName;
+    private final File mLogFile;
+    private final BufferedWriter mBufferedWriter;
+    private final ILogFormatter mLogFormatter;
+
+    /**
+     * Constructor for create a new File Appender object. This file appender will write logs to a
+     * log file with the name that is supplied via this constructor. It will throw an
+     * {@link IOException} if an error occurs while trying to create and open the log file.
+     *
+     * @param filename     the filename to use while creating log file
+     * @param logFormatter the log formatter to use while writing logs to file
+     * @throws IOException an exception that is thrown if an error occurs while creating log file
+     */
+    public FileAppender(@NonNull final String filename, @NonNull final ILogFormatter logFormatter) throws IOException {
+        mFileName = filename;
+        mLogFile = createLogFile(filename);
+        mBufferedWriter = new BufferedWriter(new FileWriter(mLogFile, true));
+        mLogFormatter = logFormatter;
+    }
+
+    @Override
+    public void append(LogLevel logLevel, String tag, String message, Throwable throwable) {
+        final String logMessage = mLogFormatter.format(logLevel, tag, message, throwable);
+        try {
+            writeUsingBufferedWriter(logMessage);
+        } catch (final IOException e) {
+            Log.e(TAG, "Error while trying to write log to file.", e);
+        }
+    }
+
+    private void writeUsingBufferedWriter(final String message) throws IOException {
+        mBufferedWriter.append(message);
+        mBufferedWriter.newLine();
```

Comment:
Just a thought - you *might* want to flush this writer every line.

Reply:
Addressed